core/amber/src/main/python/core/architecture/packaging/output_manager.py

```diff
@@ -99,6 +99,27 @@ def tuple_to_batch(
             )
         )
 
+    def state_to_batch(
+            self, state: State
+    ) -> Iterator[typing.Tuple[ActorVirtualIdentity, DataPayload]]:
+        return chain(
+            *(
+                (
+                    (
+                        receiver,
+                        (
+                            MarkerFrame(tuples)
+                            if isinstance(tuples, State)
+                            else self.tuple_to_frame(tuples)
+                        ),
+                    )
+                    for receiver, tuples in partitioner.add_state_to_batch(state)
```

Comment:
I don't understand this logic. the code is not readable to me: 
1. why `add_state_to_batch` would return `tuples`?
2. why put `tuples` into `MarkerFrame`?
3. can tuples be state? `if isinstance(tuples, State)`

Reply:
Renamed to payload. The payload can be tuples or state.